src/main/java/impl/weather/YearTemperatureStatsImpl.java

```diff
@@ -0,0 +1,62 @@
+package impl.weather;
+
+import api.weather.DayTemperatureInfo;
+import api.weather.YearTemperatureStats;
+
+import java.time.Month;
+import java.util.*;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class YearTemperatureStatsImpl implements YearTemperatureStats {
+    private final Map<Month, MonthInfo> status = new HashMap<>();
+
+    @Override
+    public void updateStats(DayTemperatureInfo info) {
+        int day = info.getDay();
+        Month month = info.getMonth();
+        int temperature = info.getTemperature();
+        status.putIfAbsent(month,new MonthInfo());
+        MonthInfo current = status.get(month);
+        current.updateAverage(temperature);
+        current.updateMaxValue(temperature);
+        status.get(month).set(day, info);
+    }
+
+    @Override
+    public Double getAverageTemperature(Month month) {
+        MonthInfo current = status.get(month);
+        if (current == null) {
+            return null;
+        }
+        return status.get(month).getAverageInfo().getAverage();
+    }
+
+    @Override
+    public Map<Month, Integer> getMaxTemperature() {
+        Map<Month, Integer> ans = new HashMap<>();
+        status.forEach((Month, MonthInfo) -> ans.put(Month, MonthInfo.getMaxTemperatureOnMonth()));
+        return ans;
+    }
+
+    @Override
+    public List<DayTemperatureInfo> getSortedTemperature(Month month) {
+        MonthInfo current;
+        current = status.get(month);
+        if (current == null) {
+            return new ArrayList<>();
+        }
+        return current.getInfo().values().stream()
+                .sorted(Comparator.comparingInt(DayTemperatureInfo::getTemperature).thenComparing(DayTemperatureInfo::getDay,Comparator.reverseOrder()))
```

Comment:
How do you consider days order with equal temp?

Reply:
As you can see the second comparator is comparing according to the day and as long as the temperature int the i + 1 day will not be entered before the temp in the i day in this case if the temp are equals in two days  then they will be sorted according to the day which mean taking the input into account

Reply:
As I see you just compare them by day number using reverse order.

Reply:
Yes exactly, because the day number and order should be identical

Reply:
Why do you think so? I can add information about 17 May before 22 May.

Reply:
Ahh ok got it I thought it is not allowed i will fix it thanks